CIP-0164/README.md

```diff
@@ -0,0 +1,2774 @@
+seen, either when it is received for the memory pool or when it is fetched after
+first being seen in an RB or EB; they perform the "reapply" operation when a
+block is being generated or validated. A more nuanced model of CPU usage in the
+simulators would account for Plutus execution explicitly, but the linear models
+described above are used to account for Plutus workloads implicitly. The
+following plot of simulation results limit each node to 4 vCPU cores and suggest
+that workloads of 20,000e9 Plutus execution steps per EB may be feasible: this
+is 1000 times the current Cardano mainnet limit of 20e9 steps for Praos blocks.
+The subsequent plot shows the 4 vCPUs becoming progressively more saturated with
+heavier Plutus execution. Although these results suggest that Leios's
+_block-level_ Plutus budget can safely be 5000 billion steps or more, it is
+important to remember that this is for conditions where honest nodes faithfully
+and promptly diffuse the transactions requiring the relatively expensive phase 2
+(Plutus) validation: adversarial nodes could attempt to delay diffusion of
+transactions in order to overwhelm honest nodes with the sudden arrival of many
+heavy Plutus transactions and little time to validate them all before voting
+upon the newly seen EB. Experiments with prototype Leios nodes will be necessary
+to more precisely quantify how much the Plutus budget could safely be increased.
+
+<div align="center">
+<a name="figure-17" id="figure-17"></a>
+
+![Fate of Plutus-heavy transactions in Leios](images/plutus-temporal-efficiency-bar.svg)
+
+<em>Figure 17: Fate of Plutus-heavy transactions in Leios</em>
+
+</div>
+
+<div align="center">
+<a name="figure-18" id="figure-18"></a>
+
+|                                                                                             |                                                                                             |
+| ------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------- |
+| ![Mean CPU usage in Plutus-heavy workloads for Leios](images/plutus-cpu-mean-histogram.svg) | ![Peak CPU usage in Plutus-heavy workloads for Leios](images/plutus-cpu-peak-histogram.svg) |
+
+<em>Figure 18: CPU usage in Plutus-heavy workloads for Leios</em>
+
+</div>
+
+In summary, Leios will require a modest increase of the [recommended hardware
+requirements][spohw]: a four-core machine will be required, but a network
+upgrade will not be needed, as 10 Mb/s is well below the bandwidth of standard
+network connections. At throughput much higher than 200 kB/s, network egress can
+become a significant cost for nodes hosted on some cloud-computing providers.
+The Leios simulations do not model memory or disk. With the advent of
+[UTxO-HD][utxohd], 16 GB of memory will remain be sufficient for Leios if the
```

Comment:
Perhaps the link https://ouroboros-consensus.cardano.intersectmbo.org/docs/references/miscellaneous/utxo-hd/ is more appropriate here